### ISSUE TITLE
New proposal on interfaces in Base Julia

### ISSUE THREAD

**Issue by gdalle** [rafaqz/Interfaces.jl#54]
Hey @rafaqz, have you seen this?
https://hackmd.io/BbEw0_B4Q8uDSS34LOvpCw

**Comment by rafaqz** [comment]
Yeah that's amazing. I saw discussion on slack but I'm really in PhD mode now so haven't looked.

But just gave it a read. Seems pretty similar to what you can do here already, like checking all of the interfaces for a package. And a similar anonymous function approach but with macro magic so its just checking methods and return  types. Looks good to me.

I hope it happens and we don't have to work on this!
